src/opm/parser/eclipse/EclipseState/Schedule/Well/WellTestState.cpp

```diff
@@ -161,6 +172,32 @@ namespace Opm {
         return well_iter->last_test;
     }
 
+    void WellTestState::updateForNewWTEST(const Opm::WellTestConfig& config)
+    {
+        // check whether to reset based on the new WTEST request
+        for (auto& well: this->wells) {
+            if (config.has(well.name, well.reason)) {
+                const auto& well_config = config.get(well.name, well.reason);
+                if (well_config.begin_report_step > well.wtest_report_step) {
+                    // there is a new WTEST input, we should reset the counting
+                    well.num_attempt = 0;
+                    well.wtest_report_step = well_config.begin_report_step;
+                }
+                assert(well_config.begin_report_step == well.wtest_report_step);
+            } else {
+                // If there is WTEST step, due to new WTEST input, which does not specify any testing closure cause,
```

Comment:
The `WellTestConfig` object should only grow - can not remove wells from the `WTEST`configuration; i.e. it is bug if this `else` clause enters.

Reply:
> The WellTestConfig object should only grow

That is not true. In Schedule::handleWTEST(), 

``` C++
1123             for(const auto& well_name : well_names) {
1124                 if (reasons.empty())
1125                     new_config->drop_well(well_name);
```

Reply:
Hmmm - that was a convincing argument ... :smile:

Reply:
Basically, later `WTEST` might cancel previous `WTEST` setup.  `drop_well` here represent the action of `cancelling`. 

Later one, we might define 
``` c++ 
enum Reason {                 
                  NONE = 0,
                  ....
```
to avoid drop_well here.

Reply:
Ahhh - Kai that was a self-shot; why not introduce the reason `NONE` or (my preference `OFF`) now; then you can (hopefully) avoid the `-1` right away. But don't sweat it.

Reply:
> why not introduce the reason NONE or (my preference OFF) now; then you can (hopefully) avoid the -1 right away. But don't sweat it.

No, I was just being honest with you.  `-1` still has its usage for the situation that there is no WTEST specified at all. It will never enter handleWTEST function at all.  Having an `NONE` will help to avoid `drop_well`. It makes the situation that `WTEST, WTEST cancelled` very slightly different with the situation `NO WTEST at all`. 

The first situation, the well have a `NONE` reason, the second situation, the well does not exist in the WellTestConfig. 

Anyway, removing `drop_well` will be considered a refactoring of the old implementation, it is not what this PR is intended to do.

It makes the test case with multiple WTESTs for the same well straight. A lot of things will be changed with more complicated cases later one.